### PR TITLE
add unit testing for tuf key verification.

### PR DESCRIPTION
**Pull request by patflynn**
Required refactoring static verifiers factory method.  I normally have a good idea how to handle DI on the server-side, but I'm just making it up as I go here. I'm very open to changes. Eventually it might make sense to collapse the two levels of indirection between verification supplier and verification. 

Work towards #60 

Signed-off-by: Patrick Flynn <patrick@chainguard.dev>

### CODE REVIEW

**Review by loosebazooka — COMMENTED** [review]
Just to start a discussion. I think I would just add the functional interface to Verifiers

```java
  @FunctionalInterface
  public interface Supplier {
      Verifier newVerifier(PublicKey publicKey) throws NoSuchAlgorithmException;
  }
```

on a method signature like
```java
/* package private */ TufClient(Verifiers.Supplier supplier)
```
you can inject with the method reference
```java
new TufClient(Verifiers::newVerifier)
```

If we don't need to expose Verifiers.Supplier we can define that functional interface in the TUF client itself.

I might also make the TUF client have one public static factory method `newTufClient` and have the various constructors you need for testing be package private?

**Comment by patflynn** [comment]
Massive improvement! Now we're cooking with gas!

File: sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java [hunk 1]
```diff
@@ -55,7 +55,21 @@ public class TufClient {
...
+  private Verifiers.Supplier verifiers;
+
+  public TufClient(Clock clock, Verifiers.Supplier verifiers) {
+    this.clock = clock;
+    this.verifiers = verifiers;
+  }
+
+  public TufClient(Verifiers.Supplier verifiers) {
+    this(Clock.systemUTC(), verifiers);
+  }
```
Comment (loosebazooka) [inline sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java:68]: should these be public?

Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java:68]: They can be package private and still be available for tests?

**Review by loosebazooka — APPROVED** [review]
Minor stuff but LGTM.

File: sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java (hunk 1 above)
Comment (patflynn) [reply sigstore-java/src/main/java/dev/sigstore/tuf/TufClient.java:68]: I'll remove public. FYI the TufClient class at the moment is really just a giant place holder for the update logic. It's going to be refactored, and the current API is not reflective in any way of what the client API will be.

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**